lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1644,7 +1642,25 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
       unsigned FwdDeclLine = 0;
       assert(SizeInBits ==
              CI.getTargetInfo().getPointerWidth(clang::LangAS::Default));
-      return createPointerSizedStruct(Scope, Decl->getNameStr(), L.File,
+      if (Opts.DebugInfoLevel > IRGenDebugInfoLevel::ASTTypes) {
+        auto *DIType = createStructType(
+            DbgTy, Decl, ClassTy, Scope, File, L.Line, SizeInBits, AlignInBits,
+            Flags, nullptr, llvm::dwarf::DW_LANG_Swift, MangledName);
+        auto SuperClassTy = ClassTy->getSuperclass();
+        if (DIType && SuperClassTy) {
```

Comment:
When does createStructType() return a nullptr?

Reply:
It shouldn't, this is just protecting against UB in case `createStructType`'s implementation is ever changed to return a null ptr.

Reply:
I think it would be clearer to assert(DIType) here. It should catch it if someone changes the implementation and makes the intention clearer. Or the function could just return a reference..